### PR TITLE
Changes for data reading

### PR DESCRIPTION
**Pull request by vandenman**
Requires https://github.com/jasp-stats/jaspBase/pull/151

### CODE REVIEW

**Comment by vandenman** [comment]
tests can only be run after https://github.com/jasp-stats/jaspBase/pull/151 is merged

File: R/descriptives.R [hunk 1]
```diff
@@ -380,7 +377,23 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
...
-.descriptivesDescriptivesTable <- function(dataset, options, jaspResults, numberMissingSplitBy = 0) {
+.descriptivesReadData <- function(options, variables, splitName) {
+
+  makeSplit <- splitName != ""
+  dataset <- .readDataSetToEnd(columns.as.numeric = variables, columns.as.factor = if (makeSplit) splitName)
+  dataset.factors <- jaspBase::readDataSetByVariableTypes(options, "variables")
+  if (makeSplit)
+    dataset.factors[[splitName]] <- dataset[[splitName]]
+
+  allMissing <- \(x) all(is.na(x))
```
Comment (JohnnyDoorn) [inline R/descriptives.R:388]: very fancy!

**Comment by JohnnyDoorn** [comment]
Awesome stuff, no more double reading! Should we first merge https://github.com/jasp-stats/jaspBase/pull/151? Does anything need to happen before we can do so? 

And is there not a better way to fix the unit tests than pasting .type everywhere? Does that need to happen everywhere we use your new data function?

**Comment by vandenman** [comment]
> Should we first merge https://github.com/jasp-stats/jaspBase/pull/151? Does anything need to happen before we can do so?

Not on my end I think, and merging that first would be the right order (and then retriggering the tests here).

File: R/descriptives.R (hunk 1 above)
Comment (vandenman) [reply R/descriptives.R:388]: Die is er al een tijdje ;), zie https://developer.r-project.org/blosxom.cgi/R-4-1-branch/NEWS/2021/05/16#n2021-05-17. De korte versie vind ik wat fijner voor (l)apply enzo.

**Comment by JohnnyDoorn** [comment]
Shall we merge this then? I guess the unit test comment does not matter for the functionality and then we can start testing

**Pull request merged**